.github/workflows/pull-gitleaks.yaml

```diff
@@ -0,0 +1,23 @@
+name: pull-gitleaks
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened, ready_for_review]
+
+env:
+  GITLEAKS_VERSION: 8.18.2
+
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Fetch gitleaks ${{ env.GITLEAKS_VERSION }}
+        run: curl -Lso gitleaks.tar.gz https://github.com/gitleaks/gitleaks/releases/download/v${{ env.GITLEAKS_VERSION }}/gitleaks_${{ env.GITLEAKS_VERSION }}_linux_x64.tar.gz && tar -xvzf ./gitleaks.tar.gz
```

Comment:
Is there are reason why we are not using the [official gitleaks action](https://github.com/gitleaks/gitleaks-action)? The licence is also already defined in secret `GITLEAKS_LICENSE` at the organisation level. By using the official action, we could define the version of gitleaks as latest. This way we avoid the version being outdated, which I fear can easily happen with the approach proposed in this PR.

Reply:
We don't use the official action, because we want to use what is recommended by Neighbours.